gymnasium/wrappers/resize_observation.py

```diff
@@ -25,23 +25,29 @@ class ResizeObservation(gym.ObservationWrapper):
         (64, 64, 3)
     """
 
-    def __init__(self, env: gym.Env, shape: Union[tuple, int]):
+    def __init__(self, env: gym.Env, shape: Union[tuple[int, int], int]) -> None:
         """Resizes image observations to shape given by :attr:`shape`.
 
         Args:
             env: The environment to apply the wrapper
             shape: The shape of the resized observations
         """
-        super().__init__(env)
-        if isinstance(shape, int):
+        if not isinstance(env.observation_space, Box):
+            raise ValueError(f"Expected the observation space to be Box, actual type: {type(env.observation_space)}")
+        dims = len(env.observation_space.shape)
+        if not 2 <= dims <= 3:
+            raise ValueError(f"Expected the observation space to have 2 or 3 dimensions, got: {dims}")
+
+        try:
+            shape = tuple(shape)
+        except TypeError:
             shape = (shape, shape)
-        assert all(x > 0 for x in shape), shape
+        if len(shape) != 2 or not all(isinstance(x, int) and x > 0 for x in shape):
+            raise ValueError(f"Expected shape to be a 2-tuple of positive integers, got: {shape}")
 
+        super().__init__(env)
```

Comment:
This should stay at the beginning of the constructor

Reply:
I thought it would be better to do input checking before any heavy initialization (i.e. fail early), but I have changed it anyway